File: neofox/references/build_IEDB_db.R

library(seqinr)
library(tidyverse)

args <- commandArgs(trailingOnly = TRUE)
path_to_file <- args[1]
path_to_file <- "/home/franlang/refs_test/iedb"

d2 <- read_csv(paste0(path_to_file, "/tcell_full_v3.csv"), skip = 1)

# grep human MHC class I epitopes that were positive in context of infectious disease
d2.human <- d2 %>%
  filter(Name == "Homo sapiens",
         `Object Type` == "Linear peptide",
         `Process Type` == "Occurrence of infectious disease",
         `Qualitative Measure`== "Positive",
         Class == "I") %>%
  distinct(Description, .keep_all = T) %>%
  mutate(epitope_id = gsub("http://www.iedb.org/epitope/", "", `Epitope IRI`),
        #fasta_header = substr(paste(epitope_id, `Antigen Name`, `Parent Species`, sep = "_"), 1,50),
        fasta_header = epitope_id,
        seq = str_split_fixed(Description, " ", 2)[,1])

# write fasta file for DB
path_to_fasta = paste(path_to_file, "IEDB.fasta", sep =  "/")
write.fasta(sequences = as.list(d2.human$seq), names = d2.human$fasta_header, file.out = path_to_fasta)

File: neofox/references/build_IEDB_db.py
#!/usr/bin/python
import pandas as pd
import sys


input_file = sys.argv[1]
output_file = sys.argv[2]

# read IEDB input file
iedb = pd.read_csv(input_file, skiprows=1)

# filter entries
filtered_iedb = iedb[
    (iedb["Name"].isin(["Homo sapiens", "Homo sapiens (human)", "Homo sapiens Caucasian", "Homo sapiens Black"])) &
    (iedb["Object Type"] == "Linear peptide") &
    (iedb["Process Type"] == "Occurrence of infectious disease") &
    (iedb["Qualitative Measure"] == "Positive") &
    (iedb["Class"] == "I")
]

# sets values for identifiers and sequences
filtered_iedb["seq"] = filtered_iedb["Description"].transform(lambda x: x.strip())
filtered_iedb["fasta_header"] = filtered_iedb["Epitope IRI"].transform(lambda x: x.replace("http://www.iedb.org/epitope/", ""))
filtered_iedb.drop_duplicates(subset="seq", keep="last", inplace=True)

# writes output FASTA file
with open(output_file, "w") as fasta:
    for index, row in filtered_iedb.iterrows():
        fasta.write(">{header}\n".format(header=str(row["fasta_header"])))
        fasta.write("{sequence}\n".format(sequence=str(row["seq"])))
